php-checks/src/main/java/org/sonar/php/checks/NestedSwitchCheck.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.php.api.tree.statement.SwitchStatementTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+@Rule(key = "S1821")
+public class NestedSwitchCheck extends PHPVisitorCheck {
+  public static final String KEY = "S1821";
+  private static final String MESSAGE = "Refactor this code to eliminate this nested \"switch\" statement.";
+  private boolean isWithinSwitch = false;
+
+  @Override
+  public void visitSwitchStatement(SwitchStatementTree tree) {
+    if (isWithinSwitch) {
+      context().newIssue(this, tree, MESSAGE);
```

Comment:
I think we should raise the issue only on the `switch` keyword. We could also make it more understandable with a secondary location on the nesting switch keyword.